src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private K[] keyArray = (K[]) new Object[10];
+    private V[] valArray = (V[]) new Object[10];
+    private int sizeArray = 0;
+
     @Override
     public void put(K key, V value) {
+
+        for (int i = 0;i < this.sizeArray;i++) {
+            if (keyEquals(i,key)) {
+                keyArray[i] = key;
+                valArray[i] = value;
+                return;
+            }
+        }
+        keyArray[this.sizeArray] = key;
```

Comment:
```suggestion
        keyArray[sizeArray] = key;
```
якщо в тебе в методі нема такої самої змінної (нема затінення) `this` можна не використовувати